### PR TITLE
Fixes for for latest SS4 and PHP 7.1

### PR DESCRIPTION
**Pull request by sminnee**
Rolling this out to a pre-release silverstripe.com on SS 4.3 and PHP 7.1, we've noticed some issues, which are resolved in this PR.

PS: we'd be happy to assist with the maintenance of this module if you wanted to add me as a committer?

### CODE REVIEW

File: src/PardotConfig.php [hunk 1]
```diff
@@ -157,44 +180,46 @@ public static function getCampaignCode()
      * Checks current pardot api credentials
      * @return string api key if valid, empty string if non-valid
```
Comment (robbieaverill) [inline src/PardotConfig.php]: Minor: could use a phpdoc for the new paramter

File: src/PardotConfig.php [hunk 2]
```diff
@@ -157,44 +180,46 @@ public static function getCampaignCode()
...
-        }
+        $key = static::loadEncryptionKeyFromConfig();
+        return Crypto::encrypt($input_string, $key);
     }
 
     /**
      * Decrypts with a bit more complexity
-     *
-     * @since 1.1.2
+     * @param $encrypted_input_string
```
Comment (robbieaverill) [inline src/PardotConfig.php]:
```suggestion
     * @param string $encrypted_input_string
```

File: src/PardotConfig.php [hunk 3]
```diff
@@ -157,44 +180,46 @@ public static function getCampaignCode()
...
+    public static function pardot_decrypt($encrypted_input_string)
     {
-        if (function_exists('mcrypt_encrypt')) {
-            $iv_size = mcrypt_get_iv_size(MCRYPT_RIJNDAEL_256, MCRYPT_MODE_ECB);
-            $iv = mcrypt_create_iv($iv_size, MCRYPT_RAND);
-            $h_key = hash('sha256', $key, true);
-            return trim(mcrypt_decrypt(MCRYPT_RIJNDAEL_256, $h_key, base64_decode($encrypted_input_string), MCRYPT_MODE_ECB, $iv));
-        } else {
-            return base64_decode($encrypted_input_string);
+        if (!$encrypted_input_string) return null;
```
Comment (robbieaverill) [inline src/PardotConfig.php]:
```suggestion
        if (!$encrypted_input_string) {
            return null;
        }
```

**Comment by sminnee** [comment]
Appreciate it, @andrewvt! Are you happy for me to merge this PR and tag a 2.0.0-beta1 release?

**Comment by andrewvt** [comment]
Yes, that would be great.  Since you guys are doing all the heavy lifting at the moment please take the lead on PRs and tag/release creation as you see fit.  I'll just monitor activity so I know what's going on.  Thanks again.

**Pull request merged**